lib/app/components/video_preview/video_preview.dart

```diff
@@ -13,19 +13,22 @@ import 'package:visibility_detector/visibility_detector.dart';
 class VideoPreview extends HookConsumerWidget {
   const VideoPreview({
     required this.videoUrl,
+    this.videoController,
     super.key,
   });
 
   final String videoUrl;
+  final VideoPlayerController? videoController;
 
   @override
   Widget build(BuildContext context, WidgetRef ref) {
-    final controller = ref.watch(
-      videoControllerProvider(
-        videoUrl,
-        looping: true,
-      ),
-    );
+    final controller = videoController ??
+        ref.watch(
+          videoControllerProvider(
+            videoUrl,
+            looping: true,
+          ),
+        )!;
```

Comment:
There is no place where we need to send `videoController` as param, so can be removed. Also it's better to get provider from WidgetRef insted passing it as param.

Reply:
I've initialized `videoController` in the `VideoMessage` to have the ability to get video aspect ratio and limit widget size according to it.